src/components/ConfirmationModal/ConfirmationModal.tsx

```diff
@@ -30,6 +31,14 @@ export type Props = PropsWithSpread<
      * Function to perform the action prompted by the modal.
      */
     onConfirm: (event: MouseEvent<HTMLElement>) => void;
+    /**
+     * Is the confirm button loading.
+     */
+    confirmButtonLoading?: boolean;
+    /**
+     * Disable the confirm button.
+     */
+    confirmButtonDisabled?: boolean;
```

Comment:
Maybe both can be `loading` and `disabled` to stay in line with how the action button component calls them? Or do you think those will be too ambiguous?